### ISSUE TITLE
pytorch study 12주차

### ISSUE THREAD

**Issue by hyunsu-yang** [hyunsu-yang/PyTorch#14]
## 강의
- https://www.boostcourse.org/ai214/lecture/43779

## 목차 
- Lab-11-5 RNN seq2seq
- Lab-11-6 PackedSequence

**Comment by hyunsu-yang** [comment]
## Lab-11-5 RNN seq2seq

### 핵심
- RNN과 seq2Seq가 다른점은 ?
  - 인코더 아키텍처와 디코더 아키텍처의 내부는 사실 두 개의 RNN 아키텍처로 구성
  - 모든 문장을 입력(encoder)받은 후에 출력(decoder)   
![image](https://user-images.githubusercontent.com/19383191/103201233-c52a2c80-4932-11eb-8119-80a4bc8d2019.png)

### 문제
- 다음 빈칸(_?_)에 알맞는 단어를 채우시오.
```
Seq2Seq 모델은 sequence의 마지막까지 보고 처리하여야 하는 _?_에 사용하기 적합한 모델로, 
 _?_ 와 _?_ 의 구조로 되어있다.
```

- 다음은 Seq2Seq 의 구현코드 일부이다. 빈칸(_?_)에 알맞는 동일한 단어를 채우고 단어의 기능에 대해 설명하시오
```

class Encoder(nn.Module):
    def __init__(self, input_size, hidden_size):
        super(Encoder, self).__init__()
        self.hidden_size = hidden_size
        self.?= nn.?(input_size, hidden_size)
        self.gru = nn.GRU(hidden_size, hidden_size)

class Decoder(nn.Module):
    def __init__(self, hidden_size, output_size):
        super(Decoder, self).__init__()
        self.hidden_size = hidden_size
        self.?= nn.?(output_size, hidden_size)
        self.gru = nn.GRU(hidden_size, hidden_size)
        self.out = nn.Linear(hidden_size, output_size)
        self.softmax = nn.LogSoftmax(dim=1)
```

**Comment by hyunsu-yang** [comment]
## Lab-11-6 PackedSequence

### 핵심
사이즈가 다른 시퀀스 데이터를 어떻게 하면 하나의 배치로 만들수 있는지 방법은?
- PackedSequence ?
  - padding Method
![image](https://user-images.githubusercontent.com/19383191/103200324-5fd53c00-4930-11eb-9952-c18b82a9498a.png)
  - packing Method  
 ![image](https://user-images.githubusercontent.com/19383191/103200433-aa56b880-4930-11eb-9289-4266ac1925f2.png)
     -  padToken 사용하지 않아도 되 효율적인 계산이 됨.
     - 내림차순으로 정렬을 해줘야하는 단점이나 구현시 패딩에 비해 좀 더 복잡

### 문제
- 사이즈가 다른 시퀀스 데이터를 하나의 배치로 만들수 있는지 두가지 방법에 대해 설명하시오
